client/packages/core/src/queryTypes.ts

```diff
@@ -70,9 +180,36 @@ type Direction = 'asc' | 'desc';
 
 type Order = { [key: string]: Direction };
 
-type $Option<Fields extends string[]> = {
+type $Option<
```

Comment:
[to-discuss] I wonder if we need $Option. 

Afaik we always will have an `S` passed into the query now.